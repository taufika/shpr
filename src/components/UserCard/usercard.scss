@import '../../variables/colors.scss';
@import '../../variables/mixins.scss';


.user-card {
  background-color: #fff;
  width: 240px;

  .header {
    padding: 16px;
    border-bottom: 2px solid $separator-color;
    display: flex;
    justify-content: space-between;
    align-items: center;

    .driver-id {
      font-size: 12px;
      color: $separator-color;

      .id-area {
        color: $primary-color;
        margin-left: 8px;
      }
    }

    .fas {
      color: $separator-color;
    }
  }

  .body {
    padding: 16px;

    .pict-container {
      border-radius: 50%;
      overflow: hidden;
      width: 80px;
      height: 80px;
      margin-bottom: 24px;
      
      img {
        height: 100%;
      }
    }

    .label {
      font-size: 12px;
      color: rgb(146, 146, 146);
      margin-top: 16px;
    }

    .value {
      font-size: 14px;
    }
  }
}

@include small-mobile-view() {
  .user-card {
    width: 100%;

    .body {
      display: flex;
      column-gap: 12px;
      justify-content: space-between;
      align-items: flex-start;

      .pict-container {
        flex-shrink: 0;
      }

      .info-container {
        width: 80%;
      }

      .label.email, .label.dob, .value.email, .value.dob {
        display: none;
      }
    }
  }
}