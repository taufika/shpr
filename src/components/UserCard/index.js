import React from 'react';

import './usercard.scss';

export default class UserCard extends React.Component {
  composeDoB(dob) {
    const date = new Date(dob);
    const year = date.getFullYear();
    const month = date.getMonth() + 1;
    const theDate = date.getDate();
    return `${theDate.toString().padStart(2, '0')}-${month.toString().padStart(2, '0')}-${year}`;
  }

  render() {
    const { name, picture, cell, email, dob } = this.props.user;

    return <div className='user-card' data-testid='user-card'>
      <div className='header'>
        <div className='driver-id'>
          <span>Driver ID</span>
          {/* the id from API sometimes return strange number or undefined */}
          <span className='id-area'>DRV{this.props.idx + 509}</span>
        </div>
        <i className="fas fa-ellipsis-h" />
      </div>
      <div className='body'>
        <div className='pict-container'>
          <img alt='drivers profile' src={picture.medium} />
        </div>

        <section className='info-container'>
          <div className='label name'>Nama Driver</div>
          <div className='value name'>{name.first}, {name.last}</div>

          <div className='label phone'>Telepon</div>
          <div className='value phone'>{cell}</div>

          <div className='label email'>Email</div>
          <div className='value email'>{email}</div>

          <div className='label dob'>Tanggal Lahir</div>
          <div className='value dob'>{this.composeDoB(dob.date)}</div>
        </section>
      </div>
    </div>
  };
}