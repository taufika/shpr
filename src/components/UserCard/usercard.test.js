import { render, screen } from '@testing-library/react';
import UserCard from './index';


const props = {
  idx: 123,
  user: {
    name: {
      first: 'Taufik',
      last: 'Akbar',
    },
    cell: '082129136223',
    email: 'taufik.akbar7@gmail.com',
    dob: {
      date: '1994-05-17',
    },
    picture: {
      medium: 'https://some.picture.url',
    },
  },
};

describe('Component:UserCard', () => {
  it.each([
    ['Driver ID', /drv632/i],
    ['Driver name', /taufik, akbar/i],
    ['Driver Phone Number', /082129136223/i],
    ['Driver Email', /taufik.akbar7@gmail.com/i],
    ['Driver DoB', /17-05-1994/i],
  ])('Should contain %s', (_, text) => {
    render(<UserCard user={props.user} idx={props.idx}/>);
    const element = screen.getByText(text);
    expect(element).toBeInTheDocument();
  });

  it('Should match snapshot', () => {
    const {container} = render(<UserCard user={props.user} idx={props.idx}/>);
    expect(container).toMatchSnapshot();
  });
});