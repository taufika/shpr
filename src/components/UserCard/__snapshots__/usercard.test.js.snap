// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component:UserCard Should match snapshot 1`] = `
<div>
  <div
    class="user-card"
    data-testid="user-card"
  >
    <div
      class="header"
    >
      <div
        class="driver-id"
      >
        <span>
          Driver ID
        </span>
        <span
          class="id-area"
        >
          DRV
          632
        </span>
      </div>
      <i
        class="fas fa-ellipsis-h"
      />
    </div>
    <div
      class="body"
    >
      <div
        class="pict-container"
      >
        <img
          alt="drivers profile"
          src="https://some.picture.url"
        />
      </div>
      <section
        class="info-container"
      >
        <div
          class="label name"
        >
          Nama Driver
        </div>
        <div
          class="value name"
        >
          Taufik
          , 
          Akbar
        </div>
        <div
          class="label phone"
        >
          Telepon
        </div>
        <div
          class="value phone"
        >
          082129136223
        </div>
        <div
          class="label email"
        >
          Email
        </div>
        <div
          class="value email"
        >
          taufik.akbar7@gmail.com
        </div>
        <div
          class="label dob"
        >
          Tanggal Lahir
        </div>
        <div
          class="value dob"
        >
          17-05-1994
        </div>
      </section>
    </div>
  </div>
</div>
`;
